File: pinyougou_sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

/**
 * 品牌服务层接口
 * @author hongbin
 */
public interface BrandService {

    /**
     * 返回全部列表
     * @return
     */
    public List<TbBrand> findAll();


    /**
     * 查询分页列表
     * @param pageNum
     * @param pageSize
     * @return
     */
    public PageResult fandPage(int pageNum,int pageSize);

    /**
     * 增加
     * @param brand
     */
    void add(TbBrand brand);

    /**
     * 删除
     * @param ids
     */
    void delete(Long[] ids);

    /**
     * 更新
     * @param brand
     */
    void update(TbBrand brand);

    /**
     * 根据Id获取
     * @param id
     */
    TbBrand getById(Long id);

    /**
     * 分页条件查询
     * @param brand
     * @param pageNum
     * @param pageSize
     * @return
     */
    PageResult search(TbBrand brand,int pageNum,int pageSize);

    /**
     * 获取品牌列表
     * @return
     */
    List<Map> selectOptionList();
}

File: pinyougou_manager_web/src/main/webapp/js/service/brandService.js
// 定义品牌服务
app.service("brandService",function ($http) {
    this.findAll=function () {
        return $http.get('../brand/findPage.do');
    };

    this.findPage=function (page,size) {
        return $http.get('../brand/findPage.do?page='+page+'&rows='+size)
    }

    this.getById=function (id) {
        return $http.get('../brand/selectOne.do?id='+id);
    }

    this.add=function (entity) {
        return $http.post('../brand/add.do',entity)
    }

    this.update=function (entity) {
        return $http.post('../brand/update.do',entity)
    }

    this.del=function (ids) {
        return $http.get('../brand/delete.do?ids='+ids)
    }

    this.search=function (page,size,searchEntity) {
        return $http.post('../brand/search.do?page='+page+'&rows='+size,searchEntity)
    }

    this.selectOptionList=function () {
        return $http.get('../brand/selectOptionList.do');
    }

});